diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 5898b8c54bef2..d924cec372c1a 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -371,6 +371,10 @@ config("compiler") {
       }
     }
 
+    if (is_linux) {
+      cflags += [ "-fstack-clash-protection" ]
+    }
+
     if (use_lld) {
       ldflags += [ "-fuse-ld=lld" ]
       if (lld_path != "") {
